### DISCUSSION TITLE
The difference between fromlabel and tolabel

### DISCUSSION THREAD

**Discussion by xguozhao** [jgraph/drawio#3607] [Q&A]
I am using sample csv to generate a network diagram. I have encountered a challenge that I want to add interface information at both ends of the two nodes, it seems like fromlabel and tolabel can help based on what i searched. But it's not working in my code. Anyone can help me understand what's difference between fromlabel and tolabel ? and any other way can help me fix this requirement. 

{"from":"local","to":"remote","invert":true,"fromlabel":"sourceport","tolabel":"destport","style":"curved=1;endArrow=none;endFill=0;dashed=0;strokeColor=#000000;"}

local,hostname,shape,remote,sourceport
a,a,mxgraph.xxx,b,eth1,eth2
b,b,mxgraph.xxx,a,eth2,eth1

**Comment by Nyc0** [comment]
I just found the difference, but since the question was asked last year, you might have to. 
fromlabel would use data of the connector source. tolabel would use data of the connect target.

let say you have the following (not b isn't connected back to a here).

> local,hostname,shape,remote,sourceport,destport
> a,a,mxgraph.xxx,b,eth1,eth2
> b,b,mxgraph.xxx,,eth2,eth1

a connects to b. 
> fromlabel being sourceport is eth1
> tolabel being destport of the target is eth1
